service/controller/v12/context/context.go

```diff
@@ -14,24 +14,29 @@ import (
 
 type contextKey string
 
-const serviceKey contextKey = "service"
+const controllerKey contextKey = "controller"
```

Comment:
I am a bit late to the party here but I would have called the `context` package differently. Like `controllercontext` or something else to prevent the conflicts with the native context package which we import basically everywhere.

Reply:
That makes sense to me. I'll rename. I would make it a part of `controller` package, but that way we'd have cyclic import.